src/state/transactions/reducer.ts

```diff
@@ -46,6 +49,12 @@ const transactionSlice = createSlice({
         return
       }
       tx.receipt = receipt
+      if (tx.info.type === TransactionType.SWAP) {
```

Comment:
you generally try to avoid mutations in the reducer - they should be pure functions. i think the Updater component would be a better home for this

Reply:
sounds good, thanks for the feedback. i assume you meant interface/src/state/transactions/updater.tsx and not interface/src/lib/hooks/transactions/updater.tsx ? moved it over to the former and verified it works the same on my local.